caas/kubernetes/provider/k8s.go

```diff
@@ -1441,6 +1485,23 @@ func (k *kubernetesClient) jujuStatus(podPhase core.PodPhase, terminated bool) s
 	}
 }
 
+func (k *kubernetesClient) jujuWorkloadStatus(podPhase core.PodPhase, terminated bool) status.Status {
+	if terminated {
+		return status.Terminated
+	}
+	// No comparable status for Waiting.
+	switch podPhase {
+	case core.PodRunning:
+		return status.Active
+	case core.PodPending:
+		return status.Maintenance
+	case core.PodFailed:
+		return status.Blocked
```

Comment:
not sure if operator status should map to workload status.
especially, should we show `blocked` if operator provision failed

Reply:
The intent is to translate the operator pod status into something resembling the application workload status (which is where this info will ultimately end up).
Possibly we might want to do that translation at display time instead.